File: backend/src/main/java/com/jaee/service/OrderService.java
package com.jaee.service;

import com.jaee.dto.common.PageResponse;
import com.jaee.dto.order.OrderDto;
import com.jaee.entity.Order;
import com.jaee.entity.User;
import com.jaee.exception.NotFoundException;
import com.jaee.repository.OrderRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Slf4j
public class OrderService {

    private final OrderRepository orderRepository;

    @Transactional(readOnly = true)
    public PageResponse<OrderDto> getUserOrders(User user, int page, int size) {
        Pageable pageable = PageRequest.of(page, size);
        Page<Order> orderPage = orderRepository.findByUserOrderByCreatedAtDesc(user, pageable);
        return PageResponse.from(orderPage, OrderDto::fromEntity);
    }

    @Transactional(readOnly = true)
    public OrderDto getOrderById(User user, Long orderId) {
        Order order = orderRepository.findByIdAndUserWithItems(orderId, user)
                .orElseThrow(() -> new NotFoundException("Order not found"));
        return OrderDto.fromEntity(order);
    }

    @Transactional(readOnly = true)
    public OrderDto getOrderByRazorpayOrderId(String razorpayOrderId) {
        Order order = orderRepository.findByRazorpayOrderIdWithItems(razorpayOrderId)
                .orElseThrow(() -> new NotFoundException("Order not found"));
        return OrderDto.fromEntity(order);
    }
}

File: frontend/src/services/orderService.ts
import { api } from '@/lib/api'
import type { ApiResponse, PageResponse, Order } from '@/types'

export const orderService = {
  getOrders: async (page: number = 0, size: number = 10): Promise<PageResponse<Order>> => {
    const response = await api.get<ApiResponse<PageResponse<Order>>>(`/orders?page=${page}&size=${size}`)
    return response.data.data
  },

  getOrderById: async (orderId: number): Promise<Order> => {
    const response = await api.get<ApiResponse<Order>>(`/orders/${orderId}`)
    return response.data.data
  },

  getOrderByRazorpayOrderId: async (razorpayOrderId: string): Promise<Order> => {
    const response = await api.get<ApiResponse<Order>>(`/orders/razorpay/${razorpayOrderId}`)
    return response.data.data
  },
}
